client/tokenized-payment-request/payment-request.js

```diff
@@ -139,93 +139,104 @@ export default class WooPaymentsPaymentRequest {
 			} );
 		paymentRequestButtonUi.showButton( paymentRequestButton );
 
-		this.attachPaymentRequestButtonEventListeners();
-		removeAction(
-			'wcpay.payment-request.update-button-data',
-			'automattic/wcpay/payment-request'
-		);
-		addAction(
-			'wcpay.payment-request.update-button-data',
-			'automattic/wcpay/payment-request',
-			async () => {
-				const newCartData = await _self.getCartData();
-				// checking if items needed shipping, before assigning new cart data.
-				const didItemsNeedShipping =
-					_self.initialProductData?.needs_shipping ||
-					_self.cachedCartData?.needs_shipping;
-
-				_self.cachedCartData = newCartData;
-
-				/**
-				 * If the customer aborted the payment request, we need to re init the payment request button to ensure the shipping
-				 * options are re-fetched. If the customer didn't abort the payment request, and the product's shipping status is
-				 * consistent, we can simply update the payment request button with the new total and display items.
-				 */
-				if (
-					! _self.isPaymentAborted &&
-					didItemsNeedShipping === newCartData.needs_shipping
-				) {
-					paymentRequest.update( {
-						total: {
-							label: getPaymentRequestData( 'total_label' ),
-							amount: parseInt(
-								newCartData.totals.total_price,
-								10
+		if ( getPaymentRequestData( 'button_context' ) === 'pay_for_order' ) {
```

Comment:
There's a lot of indentation changes here - using https://github.com/Automattic/woocommerce-payments/pull/8946/files?w=1 to hide whitespace changes helps a lot